src/protocols/secure_channel/DefaultSessionResumptionStorage.cpp

```diff
@@ -203,9 +206,10 @@ CHIP_ERROR DefaultSessionResumptionStorage::DeleteAll(FabricIndex fabricIndex)
     {
         index.mSize -= found;
         CHIP_ERROR err = SaveIndex(index);
-        stickyErr      = stickyErr == CHIP_NO_ERROR ? err : stickyErr;
         if (err != CHIP_NO_ERROR)
         {
+            stickyErr = stickyErr == CHIP_NO_ERROR ? err : stickyErr;
```

Comment:
Previous code was wrong in a few places as it was also setting the stickyErr to "value not found" which caused yet another undesired error log at higher level - so localizing the fix to this function to avoid touching several files.

Reply:
This is the log at the caller function when stickyErr gets set to "value not found":
`[1656616392249] [4871:902304] CHIP: [CTL] Warning, failed to delete session resumption state for fabric index 0x1: ../../examples/chip-tool/config/PersistentStorage.cpp:140: CHIP Error 0x000000A0: Value not found in the persisted storage`
To avoid that, stickyErr no longer gets set for error when deleting inexistent entries.

Reply:
Yes, but we _should_ have a warning because the index is not supposed to be out of sync.  Either there is a bug here we haven't root caused, or the persistent storage contents are getting into an inconsistent state regularly. Either is bad and we want to know about it.

Reply:
So I think we identified that the prior logging was giving us useful information about code that was actually broken, right?  So shouldn't we keep logging like that?

Beyond that, what is still broken that this PR fixes?  This line still will have failed with garbage data in the .ini file:

```
err = LoadState(index.mNodes[cur], resumptionId, sharedSecret, peerCATs);
```

So even with these changes, the loop continues and cannot remove the corrupt index entry:

```
        if (err != CHIP_NO_ERROR)
        {
            if (err != CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
            {
                stickyErr = stickyErr == CHIP_NO_ERROR ? err : stickyErr;

                ChipLogError(SecureChannel,
                             "Session resumption cache deletion partially failed for fabric index %u, "
                             "unable to load node state: %" CHIP_ERROR_FORMAT,
                             fabricIndex, err.Format());
            }
            continue;
        }
```

So what is this PR changing for the particular ini dataset that broke here?  For this particular problem, I think these changes are only:

* suppressing logging
* suppressing errors emitted to the caller

We shouldn't do either of these things.  So is the complaint that chip-tool is failing on `DefaultSessionResumptionStorage::DeleteAll(FabricIndex fabricIndex)`?  Well, it DID fail.  It's chip-tool's choice to fail on this or not.  If it is believed to be inappropriate for chip-tool to treat this as fatal, then chip-tool should be changed.

Please let me know if I'm missing something.  If there is something concrete that needs to change that will make this code more robust, that's great.  But suppressing errors to the caller does not add robustness.

Reply:
As I said just above we can close this as soon as the root cause is fixed which is now fixed so closing it.